core/controller/src/main/scala/org/apache/openwhisk/core/controller/Controller.scala

```diff
@@ -167,6 +167,22 @@ class Controller(val instance: ControllerInstanceId,
     }
   }
 
+  /**
+   * Handles GET /activation URI.
+   *
+   * @return running activation
+   */
+  protected[controller] val activationStatus = {
```

Comment:
can we make it so that GracefulShutdown of the controller actor system requires in flight activations to go down to zero. I think that should simplify external monitoring to determine whether it's safe to shut down so that checking these routes aren't a necessity of the deployment monitor scripting

Reply:
Yes, that is handled by https://github.com/apache/openwhisk/pull/5338/files#diff-379956107a78f18274d3895bbbb268b964cc5f89ac3601333c4255cc0bb7632dR380

It will use the number of in-flight activations.